exercise-testing-basics.js/test/students-submissions/amir-adar.test.js

```diff
@@ -0,0 +1,119 @@
+// 🏅 Your mission is to create your first tests here 💜
+// ✅ Whenever you see this icon, there's a TASK for you
+// 💡 - This is an ADVICE symbol, it will appear nearby most tasks and help you in fulfilling the tasks
+
+const testHelper = require('../test-helpers')
+const UserService = require('../../users-service')
+const jestExtended = require('jest-extended')
+
+// ✅ TASK: Run this file tests, you should see at least see this simple test below 👇 pass
+// 💡 TIP: This is how to achieve this:
+// ⓵ Open your IDE terminal
+// ⓶ Run the command 'npm run test:dev'
+// ⓷ Within the terminal, type the letter 'p', this allows you to filter tests by the file name. Now type this file name and click 'Enter'
+// 💡Another option - If you prefer not to run tests in Watch mode, just type 'npm test -- mission'
+
+describe("First test", () => {
+  test("👶🏽 This is a playground test 🚂", () => {
+    expect(true).toBe(true);
+    expect({ a: 1, b: 2, c: 3 }).toMatchObject({ b: 2 });
+    expect({ a: 1, b: 2, c: 3 }).not.toEqual({ b: 2 });
+    expect({ a: 1, b: 2, c: 3 }).toEqual({ a: 1, b: 2, c: 3 });
+  });
+});
+
+// ✅ TASK: Wrap this simple test above with a 'describe' statement
+// 💡 TIP: Start typing 'describe' and your IDE should suggest auto-completion
+// 💡 TIP: This the syntax of describe: describe('', () => {//test comes here});
+
+// ✅ TASK: Add your first test 🎉. This test should not test real code, just choose a title and put one hard-coded assertion (expect)
+// 💡 TIP: At minimum your assertion might look like: expect(true).toBe(true)
+// 💡 Try multiple assertions type to get familiar with the expect API
+
+// ✅ TASK: Test the 'validateUser' method of the 'usersService': Ensure that when no 'name' property is provided,
+//    the response 'succeeded' property is false. Remember the fail-first principle, ensure the test fail when appropriate
+// 💡 TIP: Here's a valid user object to pass. Remove the property name from this object.
+
+describe("Validate user test", () => {
+  test("Test the validation user without name", () => {
+
+    /// Arrange
+    const userExampleWithNoName = {
+      familyName: "Beck",
+      zipCode: "32486-01",
+      address: "Moonlight road 181, Alaska",
+    };
+    const userServiceTest = new UserService();
+
+    /// Act
+    const receivedResult = userServiceTest.validateUser(userExampleWithNoName);
+
+    /// Assert
+    expect(receivedResult.succeeded).toBe(false);
+    expect(receivedResult.reasons.length).toBe(1);
```

Comment:
An idea - Focus on what matters most. Not the array length rather the right reason:

expect(receivedResult).toMatchObject(succeeded:false, reasons: ['specific-reason']);